Designer Update

ingles98 opened the pull request:
Done.

Improves canvases by enabling us to edit them. **(Also titling and authoring added)**

Also making preparations for code refactor which will enable for faster process and bigger icon sizes.

Andrew-Fall commented on code/modules/designer/designer.dm:
```diff
@@ -45,18 +46,53 @@
 	usr << browse(file("html/designer.html"), "window=designer;size=600x400;can_close=1" )
-	spawn(10)usr << output("href='?src=\ref[src];[pixel_width];[pixel_height]","designer.browser:getData")
+	sleep(5)
+	usr << output("href='?src=\ref[src];[pixel_width];[pixel_height]","designer.browser:getData")
```
to_browser(user, output(...))

Andrew-Fall commented on code/modules/designer/designer.dm:
```diff
@@ -45,18 +46,53 @@
+	sleep(5)
+	usr << output("href='?src=\ref[src];[pixel_width];[pixel_height]","designer.browser:getData")
+	if (length(title) >= 1)
+		usr << output("[title]","designer.browser:setTitle")
```
to_browser(user, output(...))

Andrew-Fall commented on code/modules/designer/designer.dm:
```diff
@@ -45,18 +46,53 @@
+				if (!color || !color.len ||color.len < 3)
+					continue
+				pixel_list[x][y] = color
+		spawn(2) usr << output("[json_encode(pixel_list)]","designer.browser:getIconAsJSON")
```
to_browser(user, output(...)) Also why is spawn needed for these?

ingles98 replied:
Simply the fact that it takes long in between the JS functions. Earlier versions i had problems with it as the order would get scrambled as some functions took more time than others, as they are called asyn it seems.
Havent tested with the newer approach. 

Also the multiple outputs are for multiple function calls with different arguments. I could indeed attempt a better implementation now as the code greatly diverged from what it was, but i only want to bother with it when i get the JSON string to be directly sent to DM

EDIT:
Oh yes i didnt know about the to_browser ss13 proc. Does it grant any benefit other than being friendlier?

Andrew-Fall replied:
Not really, it's pretty much for safety reasons. Keeps travis happy

ingles98 replied:
Dunno if its only on mobile but searching for 'To_browser' finds absolutely no reference

PsyCommando replied:
Can confirm, to_browser doesn't exist @Andrew-Fall

ingles98 replied:
I still haven't understood as to why Travis is mad about the PR.
"764 << uses (expecting exactly 762)" is the only thing I see pointing towards something off in code quality yet it doesn't tell me anything.

ingles98 committed:
Improvements to Designer

Lazy commit changes (copied from changelog heh):
  - rscadd: "You are now able to add a title to canvases and sign it if you so wish."
  - rscadd: "You are now able to edit the canvas as it will load the current picture state into the editor."
  - bugfix: "Canvas pixel width and height should have been 30, and not 31 as it was."
  - wip: "Currently working on a data bridge between the browser and DM which will substantially boost perfomance while saving the icon, enabling bigger icon sizes in the future."
